### PR TITLE
Bundles Update

### LINKED ISSUE #1457: Update Tutorials with clearer simpler examples and step-by-step introduction to concepts
https://github.com/ericspod/tutorials/blob/bundles_update/bundle/bundle_intro.ipynb

### PR DESCRIPTION
**Pull request by ericspod**
Fixes #1457.

### Description
This adds tutorials going through the process of creating bundles in their entirety. 

Notebooks to add:
- [x] Very simple intro to bundles and the file structure
- [x] Simple network bundle that actually does something with training
- [ ] Bundle wrapping pre-existing code and network definitions which would cover non-MONAI code
- [ ] Bundle wrapping non-Pytorch code, maybe?

### Checks

- [x] Avoid including large-size files in the PR.
- [x] Clean up long text outputs from code cells in the notebook.
- [x] For security purposes, please check the contents and remove any sensitive info such as user names and private key.
- [x] Ensure (1) hyperlinks and markdown anchors are working (2) use relative paths for tutorial repo files (3) put figure and graphs in the `./figure` folder
- [ ] Notebook runs automatically `./runner.sh -t <path to .ipynb file>`

### CODE REVIEW

**Review by wyli — APPROVED** [review]
thanks, the tutorials look nice, cc @mingxin-zheng @Nic-Ma is it possible to skip the cell format requirements? I feel the format is defined primarily for pure python notebooks, in this PR, many cells demonstrate CLI features.

File: bundle/02_mednist_classification.ipynb [hunk 1]
```diff
@@ -0,0 +1,608 @@
+{
```
Comment (KumoLiu) [inline bundle/02_mednist_classification.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    import os
</pre><p>I would suggest separating this import part into a <code>Setup imports</code> section and also adding a <code>Setup environment</code> section in case users don't have a proper env that can't run the notebook immediately.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/Project-MONAI/tutorials/pull/1493/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

**Review by KumoLiu — APPROVED** [review]
Thanks for the update, overall looks good to me.

File: bundle/02_mednist_classification.ipynb (hunk 1 above)
Comment (ericspod) [reply bundle/02_mednist_classification.ipynb:1]: This cell isn't meant to be run, it's just the code from the monai_101 tutorial condensed to a single cell.

Comment (ericspod) [reply bundle/02_mednist_classification.ipynb:1]: What I can do is put it in the markdown cell instead. That should get around CI check problems too.

**Comment by ericspod** [comment]
The examples use the command `tree` to show the files in a bundle, is this not present in the CI environment? That might explain the errors with the notebook tests.

**Comment by ericspod** [comment]
I had to mangle the third notebook to get an output about the command line error that was encountered. The command fails with:

```
Traceback (most recent call last):
...
  File "/opt/conda/lib/python3.8/site-packages/torch/cuda/__init__.py", line 247, in _lazy_init
    torch._C._cuda_init()
RuntimeError: The NVIDIA driver on your system is too old (found version 10010). Please update your GPU driver by downloading and installing a new version from the URL: http://www.nvidia.com/Download/index.aspx Alternatively, go to: https://pytorch.org/ to install a PyTorch version that has been compiled with your version of the CUDA driver.
```

The top line of `nvidia-smi` for the containing running the tests is `Driver Version: 418.126.02   CUDA Version: 11.6` however when I run the same image locally I get `Driver Version: 530.30.02    CUDA Version: 12.1`.

**Comment by wyli** [comment]
thanks, let me try to ask about upgrading the driver version. also is it possible to remove the related component that is causing the issue?

**Comment by ericspod** [comment]
I think the component is the network being moved onto CUDA so I could just train in software but that's not demonstrating proper practice in a clear way. The network is just `densenet121`. I would have thought this issue would come up elsewhere but it may be related to spawning a subprocess in notebook 3 to train.

**Review by vikashg — APPROVED** [review]
This is great @ericspod. Thanks.

**Comment by vikashg** [comment]
As I am working through the examples I got this error

```
(monai-test) gupta@DIGI1036734:~/disk/Tools/CodeReview/monai_bundle$ python -m monai.bundle run train     --bundle_root "$BUNDLE"     --logging_file "$BUNDLE/configs/logging.conf"     --meta_file "$BUNDLE/configs/metadata.json"     --config_file "['$BUNDLE/configs/common.yaml','$BUNDLE/configs/train.yaml']"     --max_epochs 2 
2023-09-13 10:55:39,642 - INFO - --- input summary of monai.bundle.scripts.run ---
2023-09-13 10:55:39,642 - INFO - > run_id: 'train'
2023-09-13 10:55:39,642 - INFO - > meta_file: './MedNISTClassifier_v2/configs/metadata.json'
2023-09-13 10:55:39,642 - INFO - > config_file: ['./MedNISTClassifier_v2/configs/common.yaml',
 './MedNISTClassifier_v2/configs/train.yaml']
2023-09-13 10:55:39,642 - INFO - > logging_file: './MedNISTClassifier_v2/configs/logging.conf'
2023-09-13 10:55:39,642 - INFO - > bundle_root: './MedNISTClassifier_v2'
2023-09-13 10:55:39,642 - INFO - > max_epochs: 2
2023-09-13 10:55:39,642 - INFO - ---


2023-09-13 10:55:39,643 - INFO - Setting logging properties based on config: ./MedNISTClassifier_v2/configs/logging.conf.
Traceback (most recent call last):
  File "/home/gupta/miniconda3/envs/monai-test/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/home/gupta/miniconda3/envs/monai-test/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/__main__.py", line 30, in <module>
    fire.Fire()
  File "/home/gupta/miniconda3/envs/monai-test/lib/python3.10/site-packages/fire/core.py", line 141, in Fire
    component_trace = _Fire(component, args, parsed_flag_args, context, name)
  File "/home/gupta/miniconda3/envs/monai-test/lib/python3.10/site-packages/fire/core.py", line 475, in _Fire
    component, remaining_args = _CallAndUpdateTrace(
  File "/home/gupta/miniconda3/envs/monai-test/lib/python3.10/site-packages/fire/core.py", line 691, in _CallAndUpdateTrace
    component = fn(*varargs, **kwargs)
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/scripts.py", line 684, in run
    workflow = ConfigWorkflow(
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/workflows.py", line 221, in __init__
    self.parser.read_config(f=config_file)
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/config_parser.py", line 320, in read_config
    content.update(self.load_config_files(f, **kwargs))
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/config_parser.py", line 424, in load_config_files
    for k, v in (cls.load_config_file(i, **kwargs)).items():
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/bundle/config_parser.py", line 405, in load_config_file
    return yaml.load(f, CheckKeyDuplicatesYamlLoader, **kwargs)  # type: ignore[no-any-return]
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/utils/module.py", line 436, in __getattr__
    raise self._exception
  File "/home/gupta/disk/Tools/CodeReview/MONAI/monai/utils/module.py", line 395, in optional_import
    pkg = __import__(module)  # top level module
monai.utils.module.OptionalImportError: import yaml (No module named 'yaml').

For details about installing the optional dependencies, please visit:
    https://docs.monai.io/en/latest/installation.html#installing-the-recommended-dependencies
```
I installed yaml using `conda install yaml` but the error persists.

**Comment by ericspod** [comment]
The bundles need fire and yaml to run (those using JSON only need fire). I could put that in the first cell of the notebooks like other notebooks do though the requirements are for the programs being run and not the notebook itself.

**Comment by vikashg** [comment]
Ya I have both fire and yaml installed. Let me try again

**Comment by ericspod** [comment]
The package is `pyyaml`.

**Review by vikashg — APPROVED** [review]
LGTM
